spec/features/admin/products_spec.rb

```diff
@@ -470,16 +470,96 @@
       expect("#{uri.path}?#{uri.query}").to eq spree.admin_product_images_path(product, filter)
     end
 
-    scenario "editing a product's variant unit scale", js: true do
-      product = create(:simple_product, name: 'a product', supplier: @supplier2)
+    context "editing a product's variant unit scale", js: true do
+      let!(:product) { create(:simple_product, name: 'a product', supplier: @supplier2) }
 
-      visit spree.edit_admin_product_path product
-      select 'Weight (kg)', from: 'product_variant_unit_with_scale'
-      click_button 'Update'
-      expect(flash_message).to eq('Product "a product" has been successfully updated!')
-      product.reload
-      expect(product.variant_unit).to eq('weight')
-      expect(product.variant_unit_scale).to eq(1000)
+      # assertions commented out refer to bug:
+      # https://github.com/openfoodfoundation/openfoodnetwork/issues/7180
+
+      before do
+        Spree::Config.available_units = "g,lb,oz,kg,T,mL,L,kL"
```

Comment:
Changing global state in specs can influence other test cases. We need to make sure that this test leaves the database in the same state as it was before. An example of that:

https://github.com/openfoodfoundation/openfoodnetwork/blob/525fef14b17c2d95b407d3125cd6a1f7627a447b/spec/features/admin/payment_method_spec.rb#L42-L46

In this case, you can solve it simpler:

```suggestion
        allow(Spree::Config).to receive(:available_units).and_return("g,lb,oz,kg,T,mL,L,kL")
```

This is only one line which doesn't change the database and is limited to this group of test cases.

Just to add to the confusion, we use DatabaseCleaner which tries to do this for us but it doesn't work in this case. Spree::Config has an internal cache which doesn't notice a reset in the database. So we can forget about DatabaseCleaner here and just use the stub above.